### ISSUE TITLE
globalMarket Issues and comments

### ISSUE THREAD

**Issue by BlackFox403** [chill0r/GlobalMarket#4]
My wife and I have been using globalMarket solid for the week and it is one of the best mods we use. This mod adds that extra dynamic where we can transfer goods from one savegame to another. There are a couple things that we have noted...

1. The readme file says "If you get an error message on startup, it is very likely that you will have to forward port 5740 (TCP) through your router/firewall first". This is actually incorrect- according to the source "py" for both client and server, the port used is 5752.

2. The sequence of usage is a bit vague- do you start your game first and then the client, or do you start the client and then your game? Does it make a difference?

3. One thing I have noticed... we run our own "global market" server and so I have adjusted the xml files accordingly giving the hub name, IP, passwords, etc. I launch the server and it opens waiting for connection. I load my game (client is running) and transfer to and from the global storage. I see the updates in my client appear and I also see values change in the PDA mod we use. Sometimes what happens is when I do a transfer to the global storage, we get the message ""Your update was not accepted. Wrong password for this hub or you have tried to buy/sell too much in a short time. Retry in 60 seconds". If I close the client and server, then restart the server and the client, my update works. Is it a time-out issue between client and server? I had thought the server can remain running at all times and we just connect when needed. It is also possible my XML's are not properly setup, but then again if I restart both, then everything works for the duration of the play time. My wife is on one map while I am on another and we saw transfers solid for an entire evening.

4. The folder that has the client and server... I would think it may be easier and less confusing if they were separated into two- Client and Server. This way the client only needs the "economy_data.xml" while the server relies on the "economy_data_server_%s.xml" file. This is just a suggestion to make it easier for installing. Clients don't really need the server files.

5. The chat...appears to not be working? A message can be typed, but where would it appear? There is no mention of this anywhere.

As mentioned, our compliments on a well thought out mod that has certainly given us a new dynamic to our game play. We are willing to be of any help if needed for testing if required.

Regards,

BlackFox & BlueFox

**Comment by chill0r** [comment]
Hi BlackFox,

for 1: The readme is still the old one (and as you mentioned incorrect). In the next version the port will be dynamically used and the readme changed.

2: The sequence of starting doesn't matter, the client just has to be running while you want to use the market

3: Probably you tried to sell more than 100.000 of that item during that update which won't be accepted.
    Make sure to load less on each trip, the normal ingame trailers will work fine with that, just modded trailers with more than 100.000 capacity will make problems.
If that's not the case you need to wait for the next release, a new client + server will be used which shows you more informations.

4. Will be done in the next release

5. It's bugged currently (sometimes it works, sometimes not, sometimes just for a few players). It should appear normally ingame...
Will be changed in the next release too btw.

Because of testing:
I'm probably gonna finish a client + server beta next week.
I'll announce it here and on modhoster once it's ready for testing.


Greetings
chill0r

**Comment by BlackFox403** [comment]
Hello chill0r,

Thank you for the response. Regarding item #3 it was a starter in-game trailer and we only had 2300 litres, another time it was 18oo litres. But as I said, closing the client and server, then restarting the server then client and everything was fine. I was leaning towards maybe it is a connection issue where the server drops client since there is no activity. 

However, I am working with what we have currently and we look forward to the next release. The small issues are just that- small and do not take away the mod's capability.

Thank you.

Regards

**Comment by chill0r** [comment]
The client (and server) are os independend. You just need python2.7 (currently) and start the Global_Market_Client.py with it. The exe file is just a frozen python script including the runtime

**Comment by Pizmonster** [comment]
I would suggest adding the sell prices to the pda. A mod that does this is the slow bees pack though I don't know if there will be a problem with the different multifruit setups.

**Comment by chill0r** [comment]
yeah it's definitly planned too already but i canT tell if it happens in the next release or the one after.
For the next release I concentrate more on the python server and client so that it can support way more fruits and other stuff (e.g. gui for the client, client to client private chats and so on).
The game integration itself will probably the release after that
